### ISSUE TITLE
Check request body using cy.route2()

### ISSUE THREAD

**Issue by Cesaario** [cypress-io/cypress#8695]
### Current behavior
When I use cy.wait('@Test') to check the request that was mocked using cy.route2(), it does not contain a request body.
![image](https://user-images.githubusercontent.com/17167472/94608365-b5d40e00-0273-11eb-98fd-16d272f2996e.png)

### Desired behavior
I expected a behavior similar to the one using cy.route().
![image](https://user-images.githubusercontent.com/17167472/94608539-e6b44300-0273-11eb-8dee-9b407209cb4e.png)

### Test code to reproduce
```js
/// <reference types="cypress"/>

describe('Patch Test', () => {
  before(() => {
    cy.visit('https://www.test-cors.org/');
  })

  const method = 'POST';

  it('Using cy.route2()', () => {
    cy.route2(method, '**', {}).as('Route2');

    cy.get('#client_method').select(method);
    cy.get('#client_postdata').type("test");
    cy.get('#btnSendRequest').click();

    cy.wait('@Route2').then(req => {
      console.log(req);
      cy.wrap(req.request).its('body').should('contain', 'test');
    })
  })
  it('Using cy.route()', () => {
    cy.server();
    cy.route(method, '**', {}).as('Route');

    cy.get('#client_method').select(method);
    cy.get('#client_postdata').type("test");
    cy.get('#btnSendRequest').click();

    cy.wait('@Route').then(req => {
      console.log(req);
      cy.wrap(req.request).its('body').should('contain', 'test');
    })
  })
})
```

### Versions
Cypress 5.3.0
Windows 10
Chrome 85

**Comment by flotwig** [comment]
Thanks for reporting this @Cesaario. This appears to be the "request" counterpart to the unexpected behavior reported here: #8536 

As a workaround, you can intercept the request on the browser-side, and do nothing with it. This will cause the request body to be sent to the browser.

```js
  it('Using cy.route2()', () => {
    // stub request interceptor to get body sent to driver
    cy.route2(method, '**', (req) => {}).as('Route2');

    cy.get('#client_method').select(method);
    cy.get('#client_postdata').type("test");
    cy.get('#btnSendRequest').click();

    cy.wait('@Route2').then(req => {
      console.log(req);
      cy.wrap(req.request).its('body').should('contain', 'test');
    })
  })
```

**Comment by jennifer-shehane** [comment]
Released in [`5.4.0`](https://on.cypress.io/changelog#5-4-0).

This comment thread has been locked. If you are still experiencing this issue after upgrading to
Cypress v5.4.0, please open a new issue.
